staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -435,18 +434,13 @@ func (cfgCtlr *configController) digestConfigObjects(newPLs []*flowcontrol.Prior
 
 		// if we are going to issue an update, be sure we track every name we update so we know if we update it too often.
 		currResult.updatedItems.Insert(fsu.flowSchema.Name)
-		patchBytes, err := makeFlowSchemaConditionPatch(fsu.condition)
-		if err != nil {
-			// should never happen because these conditions are created here and well formed
-			panic(fmt.Sprintf("Failed to json.Marshall(%#+v): %s", fsu.condition, err.Error()))
-		}
 		if klog.V(4).Enabled() {
 			klog.V(4).Infof("%s writing Condition %s to FlowSchema %s, which had ResourceVersion=%s, because its previous value was %s, diff: %s",
 				cfgCtlr.name, fsu.condition, fsu.flowSchema.Name, fsu.flowSchema.ResourceVersion, fcfmt.Fmt(fsu.oldValue), cmp.Diff(fsu.oldValue, fsu.condition))
 		}
 		fsIfc := cfgCtlr.flowcontrolClient.FlowSchemas()
-		patchOptions := metav1.PatchOptions{FieldManager: cfgCtlr.asFieldManager}
-		_, err = fsIfc.Patch(context.TODO(), fsu.flowSchema.Name, apitypes.StrategicMergePatchType, patchBytes, patchOptions, "status")
+		applyOptions := metav1.ApplyOptions{FieldManager: cfgCtlr.asFieldManager}
+		_, err := fsIfc.ApplyStatus(context.TODO(), flowcontrolapplyconfiguration.FlowSchema(fsu.flowSchema.Name), applyOptions)
```

Comment:
.WithStatus() [or sth like that] seems to be missing

Reply:
the test build failed, i fixed it. looks like this change is causing tests to flake, will continue to debug

Reply:
The current code is patching an empty status. 

The flowcontrolapplyconfiguration is not setting any status, so it can't be working IIUC.

Reply:
oh i see what you are saying, i populated the apply configuration with status values, the controller test still fails.